src/main/java/org/terasology/servermotd/ServerMOTDSystem.java

```diff
@@ -16,30 +16,55 @@
 package org.terasology.servermotd;
 
 import org.terasology.context.Context;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
 import org.terasology.entitySystem.systems.BaseComponentSystem;
 import org.terasology.entitySystem.systems.RegisterMode;
 import org.terasology.entitySystem.systems.RegisterSystem;
-import org.terasology.network.NetworkMode;
-import org.terasology.network.NetworkSystem;
+import org.terasology.logic.console.commandSystem.annotations.Command;
+import org.terasology.logic.console.commandSystem.annotations.CommandParam;
+import org.terasology.logic.console.commandSystem.annotations.Sender;
 import org.terasology.registry.CoreRegistry;
-import org.terasology.registry.Share;
+import org.terasology.registry.In;
+
+
+@RegisterSystem(RegisterMode.AUTHORITY)
+public class ServerMOTDSystem extends BaseComponentSystem {
+    @In
+    private EntityManager entityManager;
 
-@Share(ServerMOTD.class)
-@RegisterSystem(RegisterMode.ALWAYS)
-public class ServerMOTDSystem extends BaseComponentSystem implements ServerMOTD {
     private Context context = CoreRegistry.get(Context.class);
 
-    @Override
+    private EntityRef MOTDEntity;
+    private MOTDProvider renderMOTD = new MOTDProvider();
+
     public void initialise() {
-        String motd = "default";
 
-        NetworkSystem networkSystem = context.get(NetworkSystem.class);
+    }
 
-        if (networkSystem.getMode() == NetworkMode.CLIENT) {
-            if (motd != null && motd.length() != 0) {
-                MOTDProvider renderMOTD = new MOTDProvider();
-                renderMOTD.display(motd, context);
-            }
+    @Override
+    public void postBegin() {
+        MOTDEntity = renderMOTD.getMOTDEntity(entityManager);
+    }
+
+    @Command(shortDescription = "Edit server MOTD", helpText = "Edit the current server MOTD if you are admin." +
+            " Use \'a\' - to append to current MOTD" +
+            " 'w' for a new one", runOnServer = true)
+    public String editMOTD(@CommandParam(value = "a/w") String type, @CommandParam(value = "New Message") String message, @Sender EntityRef admin) {
+        MOTDComponent comp = MOTDEntity.getComponent(MOTDComponent.class);
+        if (type.equals("a")) {
```

Comment:
My IDE is suggesting you could use a `switch` instead of the if, then you could have a `default` block for errors (or switch away to using a boolean variant command instead and not need this at all)